### ISSUE TITLE
get_transform can only be called from the main thread.

### ISSUE THREAD

**Issue by enviel** [floatinghotpot/socket.io-unity#10]
what's the best practice to set something like transform inside On event

i'm not really experienced with multithreading, from the example all i can found is "lock" which i'm not sure what put inside

```
		socket.On("pos", (data) =>
		{
			string str = data.ToString();
			Vector3 pos = JsonConvert.DeserializeObject<Vector3>(str);
			remotePlayer.transform.position = pos;
		});
```

**Comment by floatinghotpot** [comment]
Yes, the callback is running in a background thread, so not operate the UI in that.

Instead, put the data in a variable, the use the main thread to read the variable then operate the UI.

See the example:
https://github.com/floatinghotpot/socket.io-unity/blob/master/Demo/SocketIOScript.cs#L68

**Comment by enviel** [comment]
@floatinghotpot can you please update the lib to support the script that i'm doing?
just like how this lib work [https://github.com/fpanettieri/unity-socket.io-DEPRECATED](https://github.com/fpanettieri/unity-socket.io-DEPRECATED)
crating a variable for every event seems unnecessary
if i have lots of events it's going to be messy

**Comment by floatinghotpot** [comment]
I think that is same, the callback is a background thread, you can write log, but you cannot access UI.

**Comment by enviel** [comment]
@floatinghotpot i'm not accessing UI, i'm accessing transform
and i can do that in this lib [https://github.com/fpanettieri/unity-socket.io-DEPRECATED](https://github.com/fpanettieri/unity-socket.io-DEPRECATED)
i can even access UI when using that lib

**Comment by floatinghotpot** [comment]
But your suggestion is good. If initialize a function in main thread to regularly check the queue then call your callback, it should be okay.

**Comment by enviel** [comment]
yeah and that should be a core element on the lib
can you add that to the lib?

**Comment by floatinghotpot** [comment]
I think that is not difficult, just add a wrapper. I will try.

**Comment by enviel** [comment]
@floatinghotpot thankyou so much sir :)
